### PR TITLE
Phoenix.Digest mix task digests asset urls in stylesheets

### PR DESCRIPTION
**Pull request by tokafish**
This enhances the mix phoenix.digest task to replace assets found in stylesheets with their digested paths. The method is a simple regex, following the [spec](http://www.w3.org/TR/CSS2/syndata.html#value-def-uri).

A couple things to note:
* Absolute URLs found in stylesheets are not processed
* Absolute paths are assumed to be rooted in the directory being digested (with the manifest)
* Relative paths are handled
* If the asset is not found in the manifest, the stylesheet is left untouched.

Also, I thought it would be more consistent to only add this behavior to the digested version of the stylesheet. In other words, `app.css` is left unchanged, but the assets are replaced with digested versions in `app-afa653ae9d91f8a94534f8933a496bf8.css`

I feel like I could test it a bit more comprehensively if it were in a separate module, but I wanted to get overall feedback before continuing further.

See discussion [here](https://groups.google.com/forum/#!topic/phoenix-talk/5XRjFRrxlT0)

### CODE REVIEW

File: lib/phoenix/digester.ex [hunk 1]
```diff
@@ -21,11 +25,13 @@ defmodule Phoenix.Digester do
...
       unless File.exists?(output_path), do: File.mkdir_p!(output_path)
 
-      input_path
-      |> filter_files
-      |> do_compile(output_path)
-      |> generate_manifest(output_path)
-      :ok
+      digested_files = input_path
+        |> filter_files
+        |> Enum.map(&(digest(&1)))
```
Comment (josevalim) [inline lib/phoenix/digester.ex]: `&digest/1`

**Comment by josevalim** [comment]
@tokafish this looks great, thank you! Should we add a test that checks that the absolute one was not digested?

**Comment by josevalim** [comment]
@tokafish this looks great, thank you! Should we add a test that checks that the absolute one was not digested?

**Comment by chrismccord** [comment]
This is awesome! And it's build-tool agnostic which is great. We also need to make it aware of `static_url` so you can use your cdn and such.

**Comment by tokafish** [comment]
@josevalim I added the additional assertions. @chrismccord I'm not sure exactly how it would interact `static_url`, let me know what you have in mind and I'll be happy to add to the commit.

**Comment by chrismccord** [comment]
Perhaps the task can take an Endpoint as an option, but default to App.Endpoint? Then we can lookup the endpoint's static_url for that environment.

**Comment by tokafish** [comment]
I'm still getting my head around the codebase, so I'm not sure how the
defaulting would work - but once static_url was available, it would prefix
URLs in the digested stylesheet?

**Comment by tokafish** [comment]
@josevalim @chrismccord Not sure how I should proceed on this PR. Personally, I don't think using static_url is necessary. There was some discussion [here](https://groups.google.com/forum/#!topic/phoenix-talk/5XRjFRrxlT0), but it seems to me if a user wants to use a CDN, simply hosting the stylesheet from the CDN should be enough, and then the URLs within it will be interpreted properly (even if they are relative and don't specify a host).

**Comment by chrismccord** [comment]
Your are right, ignore me :)

I'll review and merge soon. Thanks!

**Comment by chrismccord** [comment]
This is so nice. Thank you!

**Pull request merged**